core/src/main/java/org/apache/iceberg/avro/GenericAvroReader.java

```diff
@@ -30,14 +30,14 @@
 import org.apache.iceberg.common.DynClasses;
 import org.apache.iceberg.data.avro.DecoderResolver;
 
-class GenericAvroReader<T> implements DatumReader<T> {
+public class GenericAvroReader<T> implements DatumReader<T> {
```

Comment:
Adding the utility methods to the Avro package would avoid needing to expose this class publicly. I think that would be good.